File: app/src/main/java/com/example/muvitracker/mainactivity/java/mylist/PrefsContract.java
package com.example.muvitracker.mainactivity.java.mylist;


import com.example.muvitracker.repo.java.dto.DetailsDto;

import java.util.List;

public interface PrefsContract {

    public interface View {

        public void updateUi (List<DetailsDto> prefsList);



    }


    public interface Presenter {

        public void loadData ();

        public void setList(List<DetailsDto> prefList); // ->fragm ->adapter

        public void toggleFavorite(DetailsDto dto);

    }

}

File: app/src/main/java/com/example/muvitracker/inkotlin/mainactivity/prefs/PrefsContract.kt
package com.example.muvitracker.inkotlin.mainactivity.prefs

import com.example.muvitracker.inkotlin.repo.dto.DetaDto

interface PrefsContract {

    interface View {

        // GET
        fun updateUi(list :List<DetaDto>)

        // SET
        // liked watched - callbacks


        // altro
        fun startDetailsFragment(movieId: Int)




    }


    interface Presenter{


        // GET
        fun getPrefsListAndUpdateUi()


        // SET
        fun toggleFovoriteItem(dtoToTogggle: DetaDto)

        fun updateWatchedItem (updatedDto: DetaDto)


        // altro
        fun onVHolderClick (movieId: Int)

    }


}